main.js

```diff
@@ -91,33 +109,104 @@ function initApp() {
   // Init BrowserWindow
   createWindow();
   setupMenu();
+}
 
-  // Load intermediary loading page
-  uiWin.loadURL(`file://${__dirname}/ui/html/loading/index.html`);
+// Show environment selection dialog
+function showSelectEnvDialog() {
+  app.focus();
+  dialog.showMessageBox({
+    type: 'question',
+    buttons: [i18n.get('mainnet'), i18n.get('testnet'), i18n.get('quit')],
+    title: i18n.get('selectQtumEnvironment'),
+    message: i18n.get('selectQtumEnvironment'),
+    defaultId: 2,
+    cancelId: 2,
+  }, (response) => {
+    switch (response) {
+      case 0: {
+        logger.info('Choose Mainnet');
 
-  // Load app main page when qtumd is fully initialized
-  server.emitter.once(ipcEvent.QTUMD_STARTED, () => {
-    uiWin.loadURL(`http://${Config.HOSTNAME}:${Config.PORT}`);
+        setQtumEnv(blockchainEnv.MAINNET);
+        initDB();
```

Comment:
await initDB() or promise.all()
migrating db step will be included in initDB() and it should be synchronously blocking

Reply:
I guess the name is confusing. I refer `initDB()` to prepare db ready for read & write. We dunt need a`initDB()` to prepare variables,  how abt moving logic in `initDB()` to `connectDB()`and rename it `initDB()`?

Reply:
`connectDB()` is called in two places. is that expected?

Reply:
looks like the purpose is just to get the DB objects. I will have `initDB` take care of initialization then export the DB datastore objs separately.